server/game/gameSystems/ViewCardSystem.ts

```diff
@@ -0,0 +1,76 @@
+import { AbilityContext } from '../core/ability/AbilityContext';
+import { BaseCard } from '../core/card/BaseCard';
+import { ViewCardType } from '../core/Constants';
+import { GameEvent } from '../core/event/GameEvent';
+import { CardTargetSystem, ICardTargetSystemProperties } from '../core/gameSystem/CardTargetSystem';
+import Player from '../core/Player';
+
+export interface IViewCardProperties extends ICardTargetSystemProperties {
+    viewType: ViewCardType;
+    sendChatMessage?: boolean;
+    message?: string | ((context) => string);
+    messageArgs?: (cards: any) => any[];
+
+    /** The player who is viewing or revealing the card. */
+    player?: Player;
+}
+
+export abstract class ViewCardSystem extends CardTargetSystem<IViewCardProperties> {
+    public override eventHandler(event, additionalProperties = {}): void {
+        const context = event.context;
+        const properties = this.generatePropertiesFromContext(context, additionalProperties);
+        if (properties.sendChatMessage) {
+            const messageArgs = this.getMessageArgs(event, context, additionalProperties);
+            context.game.addMessage(this.getMessage(properties.message, context), ...messageArgs);
+        }
+    }
+
+    public override canAffect(card: BaseCard, context: AbilityContext) {
+        // TODO: What situations would mean that a card cannot be looked at?
```

Comment:
This is interesting - I finally looked up the rules for these. "Look At" has no restrictions at all, but "Reveal" can only be cards in deck, hand, or resource zone. Neither one has any specific restrictions on faceup / facedown.

Probably worth adding a check here for the "Reveal" case on the off chance it ever becomes relevent - like for example if something triggers off a reveal, it would be important that the card could be legally revealed. Unlikely to be relevant but it's easy to add and hard to remember so we might as well do it now, imo.

Could be handled either as a check of the mode property here or as an override in the Reveal class (which would still call `super.canAffect()`